### PR TITLE
Fix repost deletion and Channel claim options

### LINKED ISSUE #6246: Repost deletion broken from pop up menu on channel page
@saltrafael aware of this one and will work on a fix

### LINKED ISSUE #6308: Don't allow editing someone else's Channel
"Edit" shouldn't appear for "Channel" that's not yours.  It populated the page with the details, but fortunately the Submit was blocked.

-

### PR DESCRIPTION
**Pull request by saltrafael**
## PR Checklist

Please check all that apply to this PR using "x":

- [ ] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [ ] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: Closes #6246, Closes #6308

- doAbandonClaim needed to Resolve the claim in order to return transaction
- channelUri wasn't properly returning the channelUri on channel claims

### CODE REVIEW

File: ui/component/claimMenuList/view.jsx [hunk 1]
```diff
@@ -158,7 +158,7 @@ function ClaimMenuList(props: Props) {
     if (!isRepost && !isChannel) {
       openModal(MODALS.CONFIRM_FILE_REMOVE, { uri });
     } else {
-      openModal(MODALS.CONFIRM_CLAIM_REVOKE, { claim, cb: !isRepost && (() => replace(`/$/${PAGES.CHANNELS}`)) });
```
Comment (tzarebczan) [inline ui/component/claimMenuList/view.jsx:161]: do you know why this was removed? I think it was to handle the re-direct after reposting.

**Comment by tzarebczan** [comment]
Sent you some feedback on Slack about the regular claim deletion issue. 

Another thing I noticed when deleting reposts is that it resolves both the original claim and repost - probably fine. 

Would be nice if the repost/claim being deleted would disappaer from the screen after deletion, but not a big deal.

**Review by jessopb — APPROVED** [review]

**Comment by saltrafael** [comment]
> Sent you some feedback on Slack about the regular claim deletion issue.

Seen!

> Another thing I noticed when deleting reposts is that it resolves both the original claim and repost - probably fine.

yep, checked and that's from doResolveUris on redux itself

> Would be nice if the repost/claim being deleted would disappaer from the screen after deletion, but not a big deal.

It should be, I'm not sure why it isn't disappearing, the claims even refresh after deleting... still figuring it out

**Comment by tzarebczan** [comment]
We'll merge this as is and you can fix that bug (happens on master too). 

The claims will continue to come up in claim search until confirmation - so if refreshed, they will come back. But maybe they can at least disappear when you delete - though not that big of a deal as long as we make this clear.

**Pull request merged**